client/blocks/post-share/connection.jsx

```diff
@@ -47,9 +49,11 @@ const PostShareConnection = ( { connection, isActive, onToggle } ) => {
 			</div>
 
 			<div className="post-share__service-account-name">
-				<span>{ external_display }</span>
+				<label id={ id }>{ external_display }</label>
 			</div>
-			<ToggleControl checked={ isActive } />
+			<BaseControl className={ classNames( 'components-toggle-control' ) }>
```

Comment:
Optional and material for follow-up: `classNames()` usage here is unnecessary, we could directly pass `"components-toggle-control" to the `className` prop.